main.js

```diff
@@ -0,0 +1,289 @@
+                        body: matter.body
+                    }, globalContext, yaml.safeLoad(matter.header)))
+                })
+                .then(html => fse.writeFile(path.join(outDir, filename), html))
+            );
+        return Promise.all(promises)
+    }).catch(err => {
+        console.log(`Error processing regular directory ${dirname}/ - ${err}`);
+    });
+}
+
+/**
+ * Renders Markdown server READMEs found in dirname, wrap with template and write to outDir/dirname/
+ * @param {string} dirname name to find Markdown source files
+ * @param {Handbars.TemplateDelegate} serverTemplate Handlebars template to make the page
+ * @param {Object} globalContext Misc context (navbar, servers, etc)
+ * @returns {Promise} wraps around a Promise.all for each file in `dirname`
+ */
+function servers(dirname, serverTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
+        /**
+         * {String[]} listing: filenames in servers/
+         */
+        let promises = listing.map(filename => 
+                fse.readFile(path.join(dirname,filename))
+                .then(content => {
+                    /**
+                     * {Buffer} content: contents of `filename`
+                     */
+                    let matter = docmatter(content.toString('UTF-8'));
+
+                    return serverTemplate(
+                        Object.assign({
+                            body: Markdown.render(matter.body)
+                        }, globalContext, yaml.safeLoad(matter.header))
+                    );
+                }).then(html => fse.writeFile(path.join(outDir, dirname, filename.replace(/\.md$/i,".html")), html))
+                
+            )
+        return fse.mkdirp(path.join(outDir, dirname)).then(Promise.all(promises));
+    }).catch(err => {
+        console.log(`Error processing server directory ${dirname}/ - ${err}`);
+    });
+}
+
+/**
+ * Formats a date "Month DD, YYYY"
+ * @param {Date} date
+ * @returns {String} 
+ */
+function formatDate(date) {
+    return date.toLocaleDateString("en-UK", {month:"long", day:"2-digit", year:"numeric"});
+}
+
+/**
+ * Generates an index of a directories PDF files, and copies them to the HTML folder.
+ * @param {string} dirname Directory to find .PDF files in
+ * @param {string} filename Filename of this page
+ * @param {Handlebars.TemplateDelegate} wrapperTemplate Handlebars template to apply to the whole page
+ * @param {Handlebars.TemplateDelegate} minuteTemplate Template to use on the listing
+ * @param {Object} globalContext Context to use for every page
+ */
+function minutes(dirname, filename, wrapperTemplate, minuteTemplate, globalContext) {
+    return fse.readdir(dirname).then(listing => {
+        let minutes = listing.filter(l => /\.pdf$/.test(l))
+                        .map(fn => {
+                            let m = /^(\d{4}-\d\d-\d\d)-([a-z0-9 ]+)\.pdf$/i.exec(fn);
```

Comment:
If a pdf file in the `minutes` directory is not in the specified format then the return on L111 will break. Consider checking the regex to make sure that the pattern matched.